src/Spatial/Euclidean/Polygon2D.cs

```diff
@@ -64,12 +64,22 @@ public Polygon2D(params Point2D[] vertices)
         /// <summary>
         /// Gets a list of vertices
         /// </summary>
-        public Point2D[] Vertices => this.points.ToArray();
+        public IEnumerable<Point2D> Vertices
+        {
+            get
+            {
+                var enumerator = this.points.GetEnumerator();
```

Comment:
Either dispose that enumerator or use a regular foreach. I'd say the latter.

Reply:
I didn't know you could yield out of a foreach ... ok will change